src/corehost/cli/args.h

```diff
@@ -21,68 +21,65 @@ struct probe_config_t
     bool only_runtime_assets;
     bool only_serviceable_assets;
 
+    bool probe_publish_dir;
+
     void print() const
     {
-        trace::verbose(_X("probe_config_t: probe=[%s] match-hash=[%d] patch-roll-forward=[%d] prerelease-roll-forward=[%d] deps-json=[%p]"),
-            probe_dir.c_str(), match_hash, patch_roll_fwd, prerelease_roll_fwd, probe_deps_json);
-    }
-    bool is_roll_fwd_set() const
-    {
-        return patch_roll_fwd || prerelease_roll_fwd;
+        trace::verbose(_X("probe_config_t: probe=[%s] match-hash=[%d] deps-json=[%p] deps-dir-probe=[%d]"),
+            probe_dir.c_str(), match_hash, probe_deps_json, probe_publish_dir);
     }
 
     probe_config_t(
         const pal::string_t& probe_dir,
         bool match_hash,
-        bool patch_roll_fwd,
-        bool prerelease_roll_fwd,
         const deps_json_t* probe_deps_json,
         bool only_serviceable_assets,
-        bool only_runtime_assets)
+        bool only_runtime_assets,
+        bool probe_publish_dir)
         : probe_dir(probe_dir)
         , match_hash(match_hash)
-        , patch_roll_fwd(patch_roll_fwd)
-        , prerelease_roll_fwd(prerelease_roll_fwd)
         , probe_deps_json(probe_deps_json)
         , only_serviceable_assets(only_serviceable_assets)
         , only_runtime_assets(only_runtime_assets)
+        , probe_publish_dir(probe_publish_dir)
     {
-        // Cannot roll forward and also match hash.
-        assert(!is_roll_fwd_set() || !match_hash);
-        // Will not roll forward within a deps json.
-        assert(!is_roll_fwd_set() || probe_deps_json == nullptr);
         // Will not do hash match when probing a deps json.
         assert(!match_hash || probe_deps_json == nullptr);
     }
 
-    static probe_config_t svc_ni(const pal::string_t& dir, bool patch_roll_fwd, bool prerelease_roll_fwd)
+    static probe_config_t svc_ni(const pal::string_t& dir)
     {
-        return probe_config_t(dir, false, patch_roll_fwd, prerelease_roll_fwd, nullptr, true, true);
+        return probe_config_t(dir, false, nullptr, true, true, false);
```

Comment:
Is specifying the exact values intended to be a simplification of what happens today?